hazelcast/src/main/java/com/hazelcast/internal/usercodedeployment/UserCodeDeploymentService.java

```diff
@@ -86,7 +92,8 @@ public ClassData getClassDataOrNull(String className) {
     public Class<?> handleClassNotFoundException(String name)
             throws ClassNotFoundException {
         if (!enabled) {
-            throw new ClassNotFoundException("User Code Deployment is not enabled. Cannot find class " + name);
+            throw new ClassNotFoundException("User Code Deployment is not enabled. Cannot find class " + name + ". "
+                    + "To deploy the class, add it to the configuration: config.getUserCodeDeploymentConfig().addClass();");
```

Comment:
I thought that `addClass` doesn't work when User Code Deployment is disabled

Reply:
Now that I deep dive, I think that the whole "is not enabled" is misleading. 

The `handleClassNotFoundException` method is only called from `UserCodeDeyplomentClassLoader` (https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/usercodedeployment/UserCodeDeploymentClassLoader.java#L57) and this classloader is only created here https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/instance/impl/Node.java#L308 which is guarded by `userCodeDeploymentConfig.isEnabled()`.

That being said, my theory is that in order to reach this error message, the user code deployment is always enabled. For that reason, I think the error message is misleading and actually just means that the class was not found, after all as the method's name also describes.

So I propose to change the wording but I guess I first need some confirmation whether my understanding is correct.

Reply:
Yes, currently it's guarded by `userCodeDeploymentConfig.isEnabled()` but I wouldn't like to make this kind of implicit assumption about its usage as it can be changed over time. Besides that, it increases the cognitive load (keeping in mind relation between `UserCodeDeploymentService`, `UserCodeDeyplomentClassLoader`, and `Node`). 

So I would suggest to leave as it is. Changes in the configuration won't do any help here.

Reply:
Ok. In that case, may I get a green tick from you? :-P

Reply:
Done. Just to be clear I meant to not change this line and keep it like this:
```java
            throw new ClassNotFoundException("User Code Deployment is not enabled. Cannot find class " + name);
```

Reply:
Ok. Updated back according to your suggestion.